### PR TITLE
[glsl-in] Add support for texture function with LOD bias

### PR DESCRIPTION
**Pull request by magcius**

### CODE REVIEW

File: src/front/glsl/functions.rs [hunk 1]
```diff
@@ -152,9 +152,12 @@ impl Program<'_> {
                         Ok(Some(args[0].0))
                     }
                     "texture" => {
-                        if args.len() != 2 {
```
Comment (JCapucho) [inline src/front/glsl/functions.rs:155]: You can write this as
```rust
if !(2..=3).contains(&args.len()) {
  return Err(ErrorKind::wrong_function_args(name, 2, args.len(), meta));
}
```

File: src/front/glsl/functions.rs [hunk 2]
```diff
@@ -152,9 +152,12 @@ impl Program<'_> {
...
-                            return Err(ErrorKind::wrong_function_args(name, 2, args.len(), meta));
-                        }
+                        let mut level = SampleLevel::Auto;
+                        match args.len() {
+                            3 => { level = SampleLevel::Bias(args[2].0); },
+                            2 => { },
+                            _ => { return Err(ErrorKind::wrong_function_args(name, 2, args.len(), meta)); }
+                        };
```
Comment (JCapucho) [inline src/front/glsl/functions.rs:160]: You can write this as
```rust
args.get(2).map(SampleLevel::Bias).unwrap_or(SampleLevel::Auto);
```

**Review by JCapucho — CHANGES_REQUESTED** [review]
We can write this in more idiomatic rust

**Comment by magcius** [comment]
Thank you! Still getting used to idiomatic Rust

**Review by JCapucho — APPROVED** [review]
Looks good now, thank you for your contribution.

**Pull request merged**